launcher/qml/org/prismlauncher/pqcstyle/PQuickPadding.h

```diff
@@ -0,0 +1,97 @@
+
+// SPDX-License-Identifier: GPL-3.0-only
+// SPDX-FileCopyrightText: 2024 Rachel Powers <508861+Ryex@users.noreply.github.com>
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2024 Rachel Powers <508861+Ryex@users.noreply.github.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *  This file incorporates work covered by the following copyright and
+ *  permission notice:
+ *
+ *    Copyright © 2017 Marco Martin <mart@kde.org>
+ *    Copyright © 2017 David Edmundson <davidedmundson@kde.org>
+ *
+ *    Licensed under LGPL-3.0-only OR GPL-2.0-only OR
+ *    GPL-3.0-only OR LicenseRef-KFQF-Accepted-GPL OR
+ *    LicenseRef-Qt-Commercial
+ *      
+ *          https://community.kde.org/Policies/Licensing_Policy
+ */
+
+#pragma once
+
+#include <QObject>
+
+#include <qqmlregistration.h>
+#include <qtmetamacros.h>
+
+class PQuickPadding : public QObject {
+    Q_OBJECT
+    QML_ANONYMOUS
+
+    Q_PROPERTY(int left READ left WRITE setLeft NOTIFY leftChanged)
+    Q_PROPERTY(int top READ top WRITE setTop NOTIFY topChanged)
+    Q_PROPERTY(int right READ right WRITE setRight NOTIFY rightChanged)
+    Q_PROPERTY(int bottom READ bottom WRITE setBottom NOTIFY bottomChanged)
+
+    int m_left;
+    int m_top;
+    int m_right;
+    int m_bottom;
+
+   public:
+    PQuickPadding(QObject* parent = nullptr) : QObject(parent), m_left(0), m_top(0), m_right(0), m_bottom(0) {}
+
+    int left() const { return m_left; }

```

Comment:
For Qproperties isn't there a macro that can generate the getters and setters.
As from me the code looks similar

type propname(){return m_propname;}
void setPropName(type propname){if (m_propname!=propname){m_propname=propname;Q_EMIT propnameChanged();}}

Reply:
If I only want to interact with the property in QML then using `Q_PROPERTY(MEMBER m_prop ...)` would work, however since I want these methods accessible form C++ I need to specify them manually. `Q_PROPERTRY`'s autogenerated accessors are only on the meta object.